lib/evmone/instructions.hpp

```diff
@@ -91,75 +91,82 @@ inline bool check_memory(ExecutionState& state, const uint256& offset, const uin
 namespace instr
 {
 
-inline StopToken stop(ExecutionState& /*state*/) noexcept
+/// The "core" instruction implementations.
+///
+/// These are minimal EVM instruction implementations which assume:
+/// - the stack requirements (overflow, underflow) have already been checked,
+/// - the "base" gas const has already been charged,
+/// - the `stack` pointer points to the EVM stack top element.
+/// Moreover, these implementations _do not_ inform about new stack height
+/// after execution. The adjustment must be performed by the caller.
+namespace core
 {
-    return {EVMC_SUCCESS};
+inline void noop(intx::uint256* /*stack*/) noexcept {}
+inline constexpr auto pop = noop;
+inline constexpr auto jumpdest = noop;
+
+template <evmc_status_code Status>
+inline StopToken stop_impl() noexcept
+{
+    return {Status};
 }
+inline constexpr auto stop = stop_impl<EVMC_SUCCESS>;
+inline constexpr auto invalid = stop_impl<EVMC_INVALID_INSTRUCTION>;
 
-inline void add(ExecutionState& state) noexcept
+inline void add(intx::uint256* stack) noexcept
```

Comment:
I previously had `StackController` which kept stack pointer and stack height and had the same API as `Stack`. Maybe it will be back in some place, but I'm not sure yet.

The `StackView` seems good idea, but mostly for keeping old code unchanged. It would work similar to `reverse_iterator`, but I'm not sure if keeping it simple is not better.